File: ConnectionData.vb
<Serializable> 
Public Class ConnectionData 

    Public StartNodeLetter As String 

    Public EndNodeLetter As String 

    Public EdgeNumber As Integer 

    Public PreviousEdgeNumber As Integer 

    Public PixelMode As Boolean 

 

    Public Sub SetStartNodeLetter(StartNodeLetter As String) 

        Me.StartNodeLetter = StartNodeLetter 

    End Sub 

    Public Sub SetEndNodeLetter(EndNodeLetter As String) 

        Me.EndNodeLetter = EndNodeLetter 

    End Sub 

    Public Sub SetEdgeNumber(EdgeNumber As Integer) 

        Me.EdgeNumber = EdgeNumber 

    End Sub 

    Public Sub SetPreviousEdgeNumber(PreviousEdgeNumber As Integer) 

        Me.PreviousEdgeNumber = PreviousEdgeNumber 

    End Sub 

    Public Sub SetPixelMode(PixelMode As Boolean) 

        Me.PixelMode = PixelMode 

    End Sub 

    Public Function GetStartNodeLetter() As String 

        Return Me.StartNodeLetter 

    End Function 

    Public Function GetEndNodeLetter() As String 

        Return Me.EndNodeLetter 

    End Function 

    Public Function GetEdgeNumber() As Integer 

        Return Me.EdgeNumber 

    End Function 

    Public Function GetPreviousEdgeNumber() As Integer 

        Return Me.PreviousEdgeNumber 

    End Function 

    Public Function GetPixelMode() As Boolean 

        Return Me.PixelMode 

    End Function 

End Class 

File: Models/ConnectionData.cs
using System;

namespace TeachingAidMac.Models
{
    [Serializable]
    public class ConnectionData
    {
        public string StartNodeLetter { get; set; } = "";
        public string EndNodeLetter { get; set; } = "";
        public int EdgeNumber { get; set; }
        public int PreviousEdgeNumber { get; set; }
        public bool PixelMode { get; set; }

        public void SetStartNodeLetter(string startNodeLetter)
        {
            StartNodeLetter = startNodeLetter;
        }

        public void SetEndNodeLetter(string endNodeLetter)
        {
            EndNodeLetter = endNodeLetter;
        }

        public void SetEdgeNumber(int edgeNumber)
        {
            EdgeNumber = edgeNumber;
        }

        public void SetPreviousEdgeNumber(int previousEdgeNumber)
        {
            PreviousEdgeNumber = previousEdgeNumber;
        }

        public void SetPixelMode(bool pixelMode)
        {
            PixelMode = pixelMode;
        }

        public string GetStartNodeLetter()
        {
            return StartNodeLetter;
        }

        public string GetEndNodeLetter()
        {
            return EndNodeLetter;
        }

        public int GetEdgeNumber()
        {
            return EdgeNumber;
        }

        public int GetPreviousEdgeNumber()
        {
            return PreviousEdgeNumber;
        }

        public bool GetPixelMode()
        {
            return PixelMode;
        }
    }
}
